Update Klog dependency

rf232 opened the pull request:
**What type of PR is this?**

/kind cleanup
/kind feature

**What this PR does / why we need it**:
Updating Klog to pickup changes needed for log sanitization.


**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1753-logs-sanitization
```
/milestone v1.20

serathius commented:
/assign @smarterclayton

serathius commented:
Looks like klog added new flag `one-output` 
We need to update tests: 
* `staging/src/k8s.io/component-base/cli/globalflag/go_default_test`
* `staging/src/k8s.io/legacy-cloud-providers/azure/go_default_test`

serathius commented:
/cc @dims

dims commented:
@rf232 @serathius there's a new flag `one-output` please see the failed CI job.

rf232 commented:
> Looks like klog added new flag `one-output`
> We need to update tests:
> 
> * `staging/src/k8s.io/component-base/cli/globalflag/go_default_test`
> * `staging/src/k8s.io/legacy-cloud-providers/azure/go_default_test`

Updated flags test.  
Azure test looks to be a flake (second run of same test passes)

rf232 committed:
Update Klog dependency

Updating Klog to pickup changes needed for log sanitization.

(https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1753-logs-sanitization)

fedebongio commented:
/remove-sig api-machinery

rf232 commented:
/retest

serathius commented:
ping @smarterclayton

dims commented:
/approve
/lgtm
/assign @liggitt 

(for root approval)

liggitt commented on vendor/k8s.io/klog/v2/klog.go:
```diff
@@ -752,18 +772,24 @@ func (l *loggingT) printWithFileLine(s severity, logr logr.Logger, file string,
+	}
 	if loggr != nil {
-		loggr.Error(err, msg, keysAndValues)
+		loggr.Error(err, msg, keysAndValues...)
```
is this a bug fix or does this represent a signature/behavior change?

serathius replied:
This is a bugfix https://github.com/kubernetes/klog/pull/186

liggitt commented on vendor/k8s.io/klog/v2/klog.go:
```diff
@@ -752,18 +772,24 @@ func (l *loggingT) printWithFileLine(s severity, logr logr.Logger, file string,
+	}
 	if loggr != nil {
-		loggr.Info(msg, keysAndValues)
+		loggr.Info(msg, keysAndValues...)
```
same question - bugfix or behavior change?

serathius replied:
https://github.com/kubernetes/klog/pull/186

liggitt commented on vendor/k8s.io/klog/v2/klog.go:
```diff
@@ -919,18 +945,22 @@ func (l *loggingT) output(s severity, log logr.Logger, buf *buffer, file string,
-			case infoLog:
-				l.file[infoLog].Write(data)
+			if l.oneOutput {
+				l.file[s].Write(data)
```
is it possible for s to be something other than fatalLog, errorLog, warningLog or infoLog? previously, values of `s` other than those were ignored

serathius replied:
No, severity is internal alias for int32. Users of klog cannot pass it's value, it's inferred from function call `klog.Info` calls subfunction with severity `infoLog`.

This code changes is meant to change fallthrough logic. Previously we wrote to files with severtity equal lower to one passed. `one-output` is meant to change this behavior and allow writing to only files matching severity. 
https://github.com/kubernetes/klog/pull/141

liggitt commented on staging/src/k8s.io/component-base/cli/globalflag/globalflags_test.go:
```diff
@@ -57,7 +57,7 @@ func TestAddGlobalFlags(t *testing.T) {
 		{
 			// Happy case
-			expectedFlag:  []string{"add-dir-header", "alsologtostderr", "help", "log-backtrace-at", "log-dir", "log-file", "log-file-max-size", "log-flush-frequency", "logtostderr", "skip-headers", "skip-log-headers", "stderrthreshold", "v", "vmodule"},
+			expectedFlag:  []string{"add-dir-header", "alsologtostderr", "help", "log-backtrace-at", "log-dir", "log-file", "log-file-max-size", "log-flush-frequency", "logtostderr", "one-output", "skip-headers", "skip-log-headers", "stderrthreshold", "v", "vmodule"},
```
not thrilled about the global flag that is not obviously prefixed or related to klog / log output

liggitt commented:
/lgtm
/approve

@dims, can you take an issue to the klog repo to ensure any future flags get a klog or log-specific prefix?

dims commented:
@liggitt will do!
